zeebe/gateway-rest/src/main/java/io/camunda/zeebe/gateway/rest/validator/RequestValidator.java

```diff
@@ -0,0 +1,65 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH under
+ * one or more contributor license agreements. See the NOTICE file distributed
+ * with this work for additional information regarding copyright ownership.
+ * Licensed under the Camunda License 1.0. You may not use this file
+ * except in compliance with the Camunda License 1.0.
+ */
+package io.camunda.zeebe.gateway.rest.validator;
+
+import static io.camunda.zeebe.gateway.rest.validator.ErrorMessages.ERROR_MESSAGE_DATE_PARSING;
+import static io.camunda.zeebe.gateway.rest.validator.ErrorMessages.ERROR_MESSAGE_EMPTY_UPDATE_CHANGESET;
+import static io.camunda.zeebe.protocol.record.RejectionType.INVALID_ARGUMENT;
+
+import io.camunda.zeebe.gateway.protocol.rest.Changeset;
+import io.camunda.zeebe.gateway.protocol.rest.UserTaskUpdateRequest;
+import io.camunda.zeebe.gateway.rest.RestErrorMapper;
+import java.time.ZonedDateTime;
+import java.time.format.DateTimeParseException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ProblemDetail;
+
+public final class RequestValidator {
+
+  public static Optional<ProblemDetail> validateUpdateRequest(
+      final UserTaskUpdateRequest updateRequest) {
+    final List<String> violations = new ArrayList<>();
+    if (updateRequest == null
+        || (updateRequest.getAction() == null && isEmpty(updateRequest.getChangeset()))) {
+      violations.add(ERROR_MESSAGE_EMPTY_UPDATE_CHANGESET);
+    }
+    if (updateRequest != null && !isEmpty(updateRequest.getChangeset())) {
+      final Changeset changeset = updateRequest.getChangeset();
+      validateDate(changeset.getDueDate(), "due date", violations);
+      validateDate(changeset.getFollowUpDate(), "follow-up date", violations);
+    }
+    if (violations.isEmpty()) {
+      return Optional.empty();
+    }
+    return Optional.of(
+        RestErrorMapper.createProblemDetail(
+            HttpStatus.BAD_REQUEST, String.join(" ", violations), INVALID_ARGUMENT.name()));
```

Comment:
🔧 How about we make the error messages consistent (e.g. always ending with a `.`), move the validation error message joining and problem detail creation to the general validator (e.g. always joining with a whitespace), and reuse the method in all request mappers and validators? The search controllers also do their own thing, I think, which makes it all inconsistent. Can also be moved into a follow up, if you think it's too much for this PR.